internal/config/watchd/watchd_test.go

```diff
@@ -0,0 +1,209 @@
+package watchd_test
+
+import (
+	"flag"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	watchdconfig "github.com/ubuntu/adsys/internal/config/watchd"
+	"gopkg.in/yaml.v2"
+)
+
+var update bool
+
+func TestGetConfigFileFromArgs(t *testing.T) {
+	tests := map[string]struct {
+		args string
+
+		want    string
+		wantErr bool
+	}{
+		"empty args":                        {wantErr: true},
+		"no config argument":                {args: "adwatchd.exe", wantErr: true},
+		"short config argument":             {args: `adwatchd.exe -c C:\path\to\adwatchd.yml`, want: `C:\path\to\adwatchd.yml`},
+		"short config argument with quotes": {args: `adwatchd.exe -c "C:\path\to\adwatchd.yml"`, want: `C:\path\to\adwatchd.yml`},
```

Comment:
Suggestion: try `-c` with no path (to protect us against bugs in the service manager) :)